woocommerce/woocommerce
# Adding inventory advanced section with radio and text fields

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Adding all fields to Advanced section of Inventory tab. 

This includes adding the generic checkbox field (with tooltip support) and an experimental `conditional` block that will only display fields when a property of the product entity matches a given value.

Closes #37401 .

![image](https://user-images.githubusercontent.com/444632/231615766-05a3802b-5562-4644-a8c5-48a65c82738f.png)

### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout the branch and use the WCA Tester Plugin to enable the `product-block-editor` feature flag.
2. Navigate to `Products -> Add new` to see the product block editor.
3. Click the Inventory tab.
4. You should see the "Advanced" button, and when clicked it should reveal the "Restrictions" checkbox.
5. Make sure the tooltip next to the Restrictions checkbox shows the tool-tip on hover.
6. Click the "Track stock quantity for this product" toggle above.
7. You should now see the "When out of stock" and "Email me when..." fields appear in the Advanced section.
8. Smoke tests these fields, entering values and ensure the persist when saved and refreshed.
